py/desispec/scripts/proc_night.py

```diff
@@ -336,6 +337,15 @@ def proc_night(night=None, proc_obstypes=None, z_submit_types=None,
     etable, ptable = load_tables(tablenames=table_pathnames, tabletypes=table_types)
     full_etable = etable.copy()
 
+    ## Pre-populate exposure table and processing table caches of all nights
+    ## if doing cross-night redshifts
+    if z_submit_types is not None and 'cumulative' in z_submit_types:
+        ## this shouldn't need to change since we've already updated the exptab
+        read_minimal_science_exptab_cols()
+        ## this would become out of date for the current night except
+        ## write_table will keep it up to date
+        read_minimal_tilenight_proctab_cols()
```

Comment:
Is this pre-population of the cache necessary for the later code to work correctly?  In general our functions do not rely upon state tracked by other functions/classes. The intention of the original caches was to make things faster, but otherwise be an implementation detail that external functions didn't need to know about.  If these calls really are necessary, that smells fragile to maintain because functions outside of the exptab/proctab readers/writers need to know about the state of the cache and act differently.  Thoughts?

Reply:
read_minimal_tilenight_proctab_cols() isn't strictly necessary as the redshift jobs all do the same command, which would produce the cache. This just explicitly does it first. It is less clear that `read_minimal_science_exptab_cols()` is ever called without a nights argument, which means that without this command it will never cache the content and will repeatedly load files.
So this is for optimization and being explicit about what we do. It is not necessary, but it is useful in my opinion.

Reply:
Thanks for explanation.  I updated the comment as a note to my future self, but otherwise this is ok and ok to leave as-is.